File: 2021/11/10/unique_occurrences.py
"""
https://leetcode.com/problems/unique-number-of-occurrences/
Author: Andrew Jarombek
Date: 11/10/2021
"""

from typing import List


def unique_occurrences(arr: List[int]) -> bool:
    occurrences = {}

    for item in arr:
        occurrences[item] = occurrences.get(item, 0) + 1

    values = set()

    for value in occurrences.values():
        if value in values:
            return False

        values.add(value)

    return True

File: 2022/10/09/uniqueOccurrences.ts
/**
 * https://leetcode.com/problems/unique-number-of-occurrences/
 * @author Andrew Jarombek
 * @since 10/9/2022
 */

function uniqueOccurrences(arr: number[]): boolean {
    let map = new Map<number, number>();

    for (let val of arr) {
        map.set(val, (map.get(val) ?? 0) + 1);
    }

    let set = new Set<number>();

    for (let val of map.values()) {
        set.add(val);
    }

    return map.size == set.size;
}